queue/abstract/queue_state.lua

```diff
@@ -65,25 +65,21 @@ local function create_state_fiber(on_state_change_cb)
         fiber.self():name('queue_state_fiber')
         while true do
             if current == queue_state.states.WAITING then
-                local rc = pcall(box.ctl.wait_rw)
-                if rc then
-                    current = queue_state.states.STARTUP
-                    log.info('Queue state changed: STARTUP')
-                    -- Wait for maximum upstream lag * 2.
-                    fiber.sleep(max_lag() * 2)
-                    on_state_change_cb(current)
-                    current = queue_state.states.RUNNING
-                    log.info('Queue state changed: RUNNING')
-                end
+                box.ctl.wait_rw()
```

Comment:
If I understand correctly, we can keep calling with `pcall` and in case of an error write a message to the log, check if `fiber` is canceled and then throw a new error with more additional context. If this is the case, I prefer to do the extra processing.

Reply:
Changed to:

```diff
--- a/queue/abstract/queue_state.lua
+++ b/queue/abstract/queue_state.lua
@@ -85,8 +85,12 @@ local function create_state_fiber(on_state_change_cb)
                     log.info('Queue state changed: WAITING')
                 end
             end
+            -- Handle server shutdown.
+            if not pcall(fiber.testcancel) then break end
         end
     end)
+
+    log.info('Finished queue state fiber')
 end

 -- Public methods.
```